### PR TITLE
Fixed #34848 -- Ignored i18n_catalog.js file when building Django's translations catalog.

### PR DESCRIPTION
**Pull request by nessita**

### CODE REVIEW

File: tests/i18n/test_extraction.py [hunk 1]
```diff
@@ -638,6 +638,31 @@ def test_default_root_settings(self):
         _, po_contents = self._run_makemessages(domain="djangojs")
         self.assertMsgId("Static content inside app should be included.", po_contents)
 
+    def test_i18n_catalog_ignored_when_invoked_for_django(self):
+        """
+        Regression test for #34848.
+        """
```
Comment (felixxm) [inline tests/i18n/test_extraction.py:644]: This docstring is unnecessary, it doesn't provide any extra information.
```suggestion
```

File: tests/i18n/test_extraction.py [hunk 2]
```diff
@@ -638,6 +638,31 @@ def test_default_root_settings(self):
...
+        # Ensure "invoked_for_django" is True
+        os.makedirs(os.path.join("conf", "locale"))
+        # Create target file so it's actually ignored
+        i18n_catalog_js_dir = os.path.join(os.path.curdir, "views", "templates")
```
Comment (felixxm) [inline tests/i18n/test_extraction.py:648]: To add more explanation, comments should state the expected behavior and omit prefixes like "Ensure" since all tests ensure things. This guideline is from [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).

Maybe:
```suggestion
        # Create a target file in the directory recognized as
        # "invoked_for_django" to be ignored.
        os.makedirs(os.path.join("conf", "locale"))
        i18n_catalog_js_dir = os.path.join(os.path.curdir, "views", "templates")
```

File: tests/i18n/test_extraction.py [hunk 3]
```diff
@@ -638,6 +638,31 @@ def test_default_root_settings(self):
...
+        os.makedirs(i18n_catalog_js_dir)
+        open(os.path.join(i18n_catalog_js_dir, "i18n_catalog.js"), "w").close()
+
+        out, po_contents = self._run_makemessages(domain="djangojs")
+        self.assertIn(f"ignoring file i18n_catalog.js in {i18n_catalog_js_dir}", out)
+
+    def test_i18n_catalog_not_ignored_when_not_invoked_for_django(self):
+        # Ensure "invoked_for_django" is False
+        assert not os.path.exists(os.path.join("conf", "locale"))
```
Comment (felixxm) [inline tests/i18n/test_extraction.py:657]: Please don't use "bare" assertions in tests, we allow them only in extremely rare cases and, IMO, that's not the case here.
```suggestion
        self.assertIs(not os.path.exists(os.path.join("conf", "locale")), False)
```

File: tests/i18n/test_extraction.py [hunk 4]
```diff
@@ -638,6 +638,31 @@ def test_default_root_settings(self):
...
+        # Create target file so it's actually ignored
+        i18n_catalog_js_dir = os.path.join(os.path.curdir, "views", "templates")
+        os.makedirs(i18n_catalog_js_dir)
+        open(os.path.join(i18n_catalog_js_dir, "i18n_catalog.js"), "w").close()
+
+        out, po_contents = self._run_makemessages(domain="djangojs")
+        self.assertIn(f"ignoring file i18n_catalog.js in {i18n_catalog_js_dir}", out)
+
+    def test_i18n_catalog_not_ignored_when_not_invoked_for_django(self):
+        # Ensure "invoked_for_django" is False
```
Comment (felixxm) [inline tests/i18n/test_extraction.py:656]:
```suggestion
        # For directories not be recognized as "invoked_for_django", there must
        # be no "conf/locale".
```

File: tests/i18n/test_extraction.py [hunk 5]
```diff
@@ -638,6 +638,31 @@ def test_default_root_settings(self):
...
+        assert not os.path.exists(os.path.join("conf", "locale"))
+        # Create target file so it exists but is not ignored
```
Comment (felixxm) [inline tests/i18n/test_extraction.py:658]:
```suggestion
        # Create target file so it exists but is not ignored.
```

**Review by felixxm — COMMENTED** [review]
@nessita Thanks :+1: I left comments to tests.

File: tests/i18n/test_extraction.py (hunk 3 above)
Comment (nessita) [reply tests/i18n/test_extraction.py:657]: I think this has to remain a bare assert. Bare asserts, when failing, produce `E` outcomes: `Error` meaning something is off with *the test code*, something is not right, the test setup or execution failed.

But `self.assert*` methods, when failing, produce `F` outcomes: `Failure` meaning the exercised function/method being tested is not behaving as expected, the functionality is not quite right (but there is *no error* in the test execution, the test code itself is -likely- correct). 

So, in this case, the fact that "conf/locale" should not exist, is not a quality that we want to ensure from running `makemessages`. For the *test* to be correct, "conf/locale" not existing is a definitive pre-requisite for the setup of the test that needs to be True for the test to make sense.

If someone would, theoretically, in a future iteration make the "conf/locale" folder available to all tests, this bare `assert` should explode quite loudly with an `E` instead of an `F`, because the test is no longer valid.

In summary, my point is that a "bare assert" should be used when the absence of a specific condition (in this case, the non-existence of the "conf/locale" folder) is a critical pre-requisite for the test to be meaningful. If this condition is not met, the test should raise an "Error" rather than a "Failure."

Comment (felixxm) [reply tests/i18n/test_extraction.py:655]: We have only 15 "bare assert" in our test suite and hundreds of unittest's assertions that check preconditions.

Comment (nessita) [reply tests/i18n/test_extraction.py]: Changed to `assertIs`.
